nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -6,27 +6,27 @@ let
   cfg = config.services.nextcloud;
   fpm = config.services.phpfpm.pools.nextcloud;
 
-  phpPackage = pkgs.php73;
-  phpPackages = pkgs.php73Packages;
+  phpPackage = pkgs.php74.buildEnv {
+    exts = pp: with pp.exts; [
+      bcmath calendar curl exif ftp filter gd gettext gmp intl json ldap
+      mysqlnd opcache openssl pcntl pdo pdo_mysql pdo_odbc pdo_pgsql
+      pdo_sqlite pgsql readline session soap sodium sqlite3 zip zlib mbstring
+      posix ctype dom simplexml xmlreader xmlwriter pp.apcu
+      pp.redis pp.memcached pp.imagick
+    ];
+    extraConfig = phpOptionsStr;
+  };
```

Comment:
Have you confirmed that this config is actually taken into account by phpfpm entirely? The module adds another `-c php.ini` in the systemd-unit which IIRC overrides changes from `php.buildEnv`:

https://github.com/NixOS/nixpkgs/blob/24d456a48daccf1f8b179c53d92ed3c2c3bc47b4/nixos/modules/services/web-servers/phpfpm/default.nix#L276

Reply:
That's why `php.buildEnv` was rewritten to not use the `-c`-flag. It instead uses an environment variable to set additional config.